benchmark_runner/common/clouds/BM/bare_metal_operations.py

```diff
@@ -0,0 +1,284 @@
+
+    def version_already_installed(self):
+        """
+        This method validate if version is already install
+        :return: True if it already installed, False if it NOT already installed
+        """
+        if self.get_ocp_server_version() == 'null':
+            return False
+        elif self.LATEST in self._install_ocp_version and self.get_ocp_server_version() == self._get_installation_version():
+            return self.get_ocp_server_version().strip()
+        elif self._install_ocp_version == self.get_ocp_server_version():
+            return self._install_ocp_version
+        else:
+            return False
+
+    @logger_time_stamp
+    def restart_pod_ci(self):
+        """
+        This method restart pod ci: elastic, kibana, grafana, nginx and flask - solved connectivity issue after installation
+        :return:
+        """
+        self._remote_ssh.run_command('podman pod restart pod_ci')
+
+    @typechecked()
+    @logger_time_stamp
+    def update_ocp_version(self, file_name: str):
+        """
+        This method update the ocp version on provision machine
+        :param file_name:
+        :return:
+        """
+        # Get the latest assisted installer version
+        if self.LATEST in self._install_ocp_version:
+            self._install_ocp_version = self._get_installation_version()
+        openshift_version_data = self._install_ocp_version.split('.')
+        self._remote_ssh.replace_parameter(remote_path='/root/jetlag/ansible/vars',
+                                            file_name=file_name,
+                                            parameter='ocp_release_image:',
+                                            value=f'quay.io\/openshift-release-dev\/ocp-release:{self._install_ocp_version}-x86_64')
+        self._remote_ssh.replace_parameter(remote_path='/root/jetlag/ansible/vars',
+                                            file_name=file_name,
+                                            parameter='openshift_version:',
+                                            value=f'"{openshift_version_data[0]}.{openshift_version_data[1]}"')
+
+    def update_provision_config(self):
+        """
+        This method update provision config file
+        :return:
+        """
+        # update ssh config - must add it and also mount it
+        self._ssh.run(f"echo Host provision > /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'HostName {self._provision_ip} >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'User {self._user} >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'IdentityFile {self._container_private_key_path} >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'StrictHostKeyChecking no >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'ServerAliveInterval 30 >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'ServerAliveCountMax 5 >> /{self._user}/.ssh/config")
```

Comment:
Better if you can construct the file here and then scp/cat it onto the remote file.